### ISSUE TITLE
Avoid table aliases in join conditions (especially initialisms)

### ISSUE THREAD

**Issue by NiallRees** [sqlfluff/sqlfluff#469]
'Avoid table aliases in join conditions (especially initialisms) – it's harder to understand what the table called "c" is compared to "customers".'

Taken from the [dbt coding conventions](https://github.com/fishtown-analytics/corp/blob/master/dbt_coding_conventions.md#ctes)

**Comment by pwildenhain** [comment]
thinking more about this -- sometimes aliasing it required, because you need to self-join. 

Ex:
```sql
select
    my_table.a,
    my_table_again.b,
from
    my_table
    left join my_table as my_table_again on my_table.foreign_key = my_table_again.foreign_key
```
In that case you **have** to alias.

Could we make the rule smart enough to not trigger if joining to a previously declared table?

FYI @piotrgredowski for #473

**Comment by piotrgredowski** [comment]
I guess it should be possible. I'll think about it tomorrow :)

**Comment by piotrgredowski** [comment]
@pwildenhain now it should allow for self-joins too.
